src/utils/paginator.ts

```diff
@@ -0,0 +1,121 @@
+const {
+  MessageActionRow,
+  Message,
+  MessageEmbed,
+  MessageButton,
+} = require("discord.js");
+
+/**
+ * Creates a pagination embed
+ * @param {Interaction} interaction
+ * @param {MessageEmbed[]} pages
+ * @param {MessageButton[]} buttonList
+ * @param {number} timeout
+ * @returns
+ */
+const paginationEmbed = async (interaction, pages, buttonList, timeout = 120000) => {
+  if (!pages) throw new Error("No pages for paginationEmbed!.");
+  if (!buttonList) buttonList = [
+      new MessageButton()
+                    .setCustomId('tobeginning')
+                    .setStyle('PRIMARY')
+                    .setEmoji('921073324094804038'),
+      new MessageButton()
+                    .setCustomId('last')
+                    .setStyle('PRIMARY')
+                    .setEmoji('921073293572853803'),
+      new MessageButton()
+                    .setCustomId('next')
+                      .setEmoji('921073355073933313')
+                    .setStyle('PRIMARY'),
+      new MessageButton()
+                    .setCustomId('toend')
+                  .setEmoji('921073383955922995')
+                    .setStyle('PRIMARY'),
+      new MessageButton()
+                    .setCustomId('discard')
+                  .setEmoji('921074968660414485')
+                    .setStyle('DANGER'),
+  ]
+  if (buttonList[0].style === "LINK" || buttonList[1].style === "LINK")
+    throw new Error(
+      "Link buttons are not supported!"
+    );
+
+  let page = 0;
+
+  const row = new MessageActionRow().addComponents(buttonList);
+
+  //has the interaction already been deferred? If not, defer the reply.
+  if (interaction.deferred == false){
```

Comment:
There is really no need to defer the reply if we respond right after.

Reply:
Read comment below.